DotaStats/Common/CachedImageView/CachedImageView.swift

```diff
@@ -0,0 +1,85 @@
+import UIKit
+
+final class CachedImageView: UIImageView {
+    private let imageLoader = ImageServiceImp.shared
+    private let gradientLayer: CAGradientLayer = {
+        let gradientLayer = CAGradientLayer()
+        gradientLayer.startPoint = CGPoint(x: 0, y: 1)
+        gradientLayer.endPoint = CGPoint(x: 1, y: 1)
+        gradientLayer.colors = [UIColor.lightGray.cgColor,
+                                UIColor.white.cgColor,
+                                UIColor.lightGray.cgColor]
+
+        return gradientLayer
+    }()
+    private var isAnimateRunning = false
+
+    private let lastRequest: Cancellable? = nil
+
+    init() {
+        super.init(frame: CGRect(x: 50, y: 50, width: 100, height: 100))
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func loadWithUrl(_ url: String) {
```

Comment:
Может лучше поменять сигнатуру на привычную
```suggestion
    func setImage(with url: String) {
```
Кажется, что так будет достаточно просто использовать ее извне